src/AWS/CRT/Auth/AwsCredentials.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace AWS\CRT\Auth;
+
+use AWS\CRT\NativeResource as NativeResource;
+
+/**
+ * Represents a set of AWS credentials
+ *
+ * @param array options:
+ * - string access_key_id - AWS Access Key Id
+ * - string secret_access_key - AWS Secret Access Key
+ * - string session_token - Optional STS session token
+ * - int expiration_timepoint_seconds - Optional time to expire these credentials
+ */
+final class AwsCredentials extends NativeResource {
+
+    static function defaults() {
+        return array(
+            'access_key_id' => '',
+            'secret_access_key' => '',
+            'session_token' => '',
+            'expiration_timepoint_seconds' => 0,
+        );
+    }
+
+    private $access_key_id;
+    private $secret_access_key;
+    private $session_token;
+    private $expiration_timepoint_seconds = 0;
+
+    public function __get($name) {
+        return $this->$name;
+    }
+
+    function __construct(array $options = array()) {
+        parent::__construct();
+        if (count($options) == 0) {
+            $options = self::defaults();
+        }
+        $this->access_key_id = $options['access_key_id'];
```

Comment:
This will throw errors for undefined indices.  A better way to do it is to have a getter function like in our AWS client: 


 ```
   public function getConfig($option = null)
    {
        return $option === null
            ? $this->config
            : (isset($this->config[$option])
                ? $this->config[$option]
                : null);
    }
```

Also, see: https://stackoverflow.com/questions/1195549/how-to-avoid-undefined-index-errors

Reply:
Specifically this would happen in a code path where one to three options are set (so defaults don't get set), and there's a blank option

Reply:
Good catch, will do.